### PR TITLE
node-macaddr#3 Want custom separator and custom formatter

### PR DESCRIPTION
**Pull request by cableray**
In order to output various types easily, allow setting a custom separator, and allow providing a function to specify formatting.

Let me know what else is needed, please :)

### CODE REVIEW

**Comment by cableray** [comment]
I think I'm going to change it so that the formatter doesn't override the padding, and then extract the default formatter to a function to simplify...

**Comment by cableray** [comment]
I'd like to change the loop to a map, but that may be too much for this pr....

**Comment by bahamat** [comment]
Still evaluating this, but at the very least it will need a version bump in package.json.

**Comment by cableray** [comment]
> Still evaluating this, but at the very least it will need a version bump in package.json.

So, `0.1.0`? It's a non-breaking new feature, so minor bump? Given as it is currently at `0.0.1`, I didn't think it had been versioned before... Also, should I commit the auto reformatting of `package.json`? (Looks like npm re-ordered some packages)

**Comment by bahamat** [comment]
Yeah, 0.1.0 is fine. There's only 3 commits so far in this repo so far. The first one was just creating the repo from our template, the second was ver 0.0.1, and the third commit was just updating the `make check` dependency, and no change in the behavior of the library.

And for the structure of `package.json`, making the diff minimal is easier to read, but other than that I don't have a strong opinion about it.

File: README.md [hunk 1]
```diff
@@ -35,6 +35,10 @@ object allows for controlling the output, and takes the following fields:
 
 - `zeroPad` (default: `true`), whether to pad octets that could print as a
   single character with a `0`.
+- `separator` (default: `':'`), separator between octets.
+- `octetFormatter` (default: (none)), a function that takes the integer 
+  octet as the argument and returns the formatted octet. If `zeroPad` is
```
Comment (bahamat) [inline README.md]: Whitespace at end of lines 39 & 40 should be removed.

File: test/output.test.js [hunk 2]
```diff
@@ -56,7 +56,69 @@ test('output without leading zeros', function (t) {
     strs.forEach(function (str) {
         var addr = macaddr.parse(str[0]);
         t.equal(addr.toString({ zeroPad: false }), str[1],
-            str[0] + ' (with zeroPad=false)');
+          str[0] + ' (with zeroPad=false)');
```
Comment (bahamat) [inline test/output.test.js]: Indenting is not correct here.

**Review by bahamat — CHANGES_REQUESTED** [review]
Ok, overall, things look pretty good with this. Only some small nits on whitespace.

Ensure that `make check` passes. After these and the version bump in package.json we'll be good to merge.

**Comment by cableray** [comment]
Oops, changed the spacing in brackets... let me revert that...

**Review by bahamat — APPROVED** [review]

**Pull request merged**